golang/Solutions/assgn1.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+var file string = "history.txt"
+
+func check(e error) {
+	if e != nil {
+		os.Truncate(file, 0)
+		panic(e)
+	}
+}
+
+func executeCmd(cmd string) string {
+	out, err := exec.Command(cmd).Output()
+	check(err)
+	return strings.TrimSuffix(string(out), "\n")
+}
+
+func addToHistory(cmd string) {
+	// write to file
+	f, err := os.OpenFile(file, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	check(err)
+	_, err = f.Write([]byte(cmd + "\n"))
+	check(err)
+	f.Close()
+}
+
+func main() {
+
+	var header string
+	var cmd string
+
+	for {
+		header = ""
+		header += executeCmd("whoami") + "/" + executeCmd("hostname") + executeCmd("pwd") + "~"
```

Comment:
Use other special characters instead of `/`. It will be confusing with the pwd value